crates/runtime/tests/models/s3_vectors.rs

```diff
@@ -29,74 +35,136 @@ mod search {
+        test_dataset.vectors = Some(new_s3_vector_store(
+            "spice-ci-tests-s3-vectors-basic",
+            &index_name,
+        ));
 
         let app = AppBuilder::new("search_app")
             .with_dataset(test_dataset)
-            .with_embedding(titan_embeddings)
+            .with_embedding(get_huggingface_embeddings(
+                "sentence-transformers/all-MiniLM-L6-v2",
+                "hf_minilm",
+            ))
             .build();
 
         let rt = start_app(app).await?;
 
         run_and_snapshot_query(
             &rt,
-            "SELECT i_item_sk, i_item_desc, round(score, 2) FROM vector_search(item, 'Patient') where i_item_sk > 5 order by score desc LIMIT 4;",
+            "SELECT i_item_sk, i_item_desc, round(score, 1) FROM vector_search(item, 'Patient') where i_item_sk > 5 order by score desc LIMIT 4;",
             "basic",
         )
         .await?;
 
         run_and_snapshot_query(
             &rt,
-            "explain SELECT i_item_sk, i_item_desc, round(score, 2) FROM vector_search(item, 'Patient') where i_item_sk > 5 order by score desc LIMIT 4;",
+            "explain SELECT i_item_sk, i_item_desc, round(score, 1) FROM vector_search(item, 'Patient') where i_item_sk > 5 order by score desc LIMIT 4;",
             "basic_explain",
         )
         .await?;
 
         Ok(())
     }
 
+    #[tokio::test]
+    async fn s3_vectors_filters_pushdown() -> Result<(), anyhow::Error> {
+        for env_var in ["AWS_S3_VECTORS_KEY", "AWS_S3_VECTORS_SECRET"] {
+            verify_env_secret_exists(env_var)
+                .await
+                .map_err(anyhow::Error::msg)?;
+        }
+
+        let _tracing = crate::init_tracing(DEFAULT_TRACING_MODELS);
+
+        let mut test_dataset = get_package_delivery_dataset("delivery", None, "hf_minilm");
+
+        // Generate a unique index name for each test run
+        let index_name = format!("test-index-{}", rand::random::<u8>() % 11);
+
+        test_dataset.vectors = Some(new_s3_vector_store(
+            "spice-ci-tests-s3-vectors-filters-pushdown",
+            &index_name,
+        ));
+
+        let app = AppBuilder::new("search_app")
+            .with_dataset(test_dataset)
+            .with_embedding(get_huggingface_embeddings(
+                "sentence-transformers/all-MiniLM-L6-v2",
+                "hf_minilm",
+            ))
+            .build();
+
+        let rt = start_app(app).await?;
+
+        // Failed sms notifications on heavy deliveries sent to the wrong location"
+        run_and_snapshot_query(
+            &rt,
+            r#"
+            SELECT 
+              "message.body",
+              attempt_count, "message.status",
+              package_weight_kg,
+              round(score, 1)
+            FROM vector_search(delivery, 'wrong location')
+            WHERE attempt_count > 1 AND package_weight_kg > 5.0 AND "message.status"='FAILED'
+            ORDER BY package_weight_kg desc, score DESC
```

Comment:
We order on package_weight_kg and not on score as we have multiple items with similar body so score and order of records may fluctuate across test runs

Reply:
nit, should be comment for the future